Esri/arcgis-maps-sdk-swift-samples
# [New] Edit features with feature linked annotation

## Description

This PR implements `Edit features with feature-linked annotation` in `Edit and Manage Data` category.
URL to README: [README](https://github.com/Esri/arcgis-maps-sdk-swift-samples/tree/Caleb/New-EditFeaturesWithFeatureLinkedAnnotation/Shared/Samples/Edit%20features%20with%20feature-linked%20annotation)

## Linked Issue(s)

- `swift/issues/5377`

## How To Test

- Tap on a point to edit it's address and tap again to move the point.
- Tap on a straight polyline and tap again to move it's nearest vertex.
- See the annotations update accordingly.

## Screenshots


![edit-features-with-feature-linked-annotation](https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/34120440/18ce9926-d847-4464-9b35-09db123241a5)
